docs/grammar.md

```diff
@@ -94,11 +95,13 @@ ifCondition -> "if" parenthesizedExpression object
 
 forExpression -> "[" "for" IDENTIFIER "in" expression ":" expression(body) "]"
 
-interpString ->  interpStringLeftPiece ( expression interpStringMiddlePiece )* expression interpStringRightPiece | literalString
-interpStringLeftPiece -> "'" STRINGCHAR* "${"
-interpStringMiddlePiece -> "}" STRINGCHAR* "${"
-interpStringRightPiece -> "}" STRINGCHAR* "'"
-literalString -> "'" STRINGCHAR* "'"
+interpString ->  stringLeftPiece ( expression stringMiddlePiece )* expression stringRightPiece | stringComplete
+stringLeftPiece -> "'" STRINGCHAR* "${"
+stringMiddlePiece -> "}" STRINGCHAR* "${"
+stringRightPiece -> "}" STRINGCHAR* "'"
+stringComplete -> "'" STRINGCHAR* "'"
```

Comment:
>STRINGCHAR [](start = 22, length = 10)

Not a new issue and definitely non-blocking for this PR, but I just realized we haven't defined what STRINGCHAR is or even IDENTIFIER.

Reply:
Yes :) I decided to leave it along with this PR